src/wallet/wallet.h

```diff
@@ -916,10 +917,9 @@ class CWalletTx : public CMerkleTx
 
     //! filter decides which addresses will count towards the debit
     CAmount GetDebit(const isminefilter& filter) const;
-    CAmount GetCredit(const isminefilter& filter, bool recalculate = false, bool fUnspent = false) const;
-    CAmount GetUnspentCredit(const isminefilter& filter) const;
+    CAmount GetCredit(const isminefilter& filter, bool recalculate = false) const;
     CAmount GetImmatureCredit(bool fUseCache = true, const isminefilter& filter = ISMINE_SPENDABLE_ALL) const;
-    CAmount GetAvailableCredit(bool fUseCache = true) const;
+    CAmount GetAvailableCredit(bool fUseCache = true, const isminefilter& filter=ISMINE_SPENDABLE) const NO_THREAD_SAFETY_ANALYSIS;
```

Comment:
Don't think we need to be setting the `NO_THREAD_SAFETY_ANALYSIS` flag here, at least my local clang doesn't complain with it removed and `-Wthread-safety-analysis` set

Reply:
updated